src/System.Private.ServiceModel/src/System/ServiceModel/Description/DataContractSerializerOperationBehavior.cs

```diff
@@ -82,7 +82,8 @@ public virtual XmlObjectSerializer CreateSerializer(Type type, string name, stri
             settings.MaxItemsInObjectGraph = MaxItemsInObjectGraph;
             settings.DataContractResolver = DataContractResolver;
             DataContractSerializer dcs = new DataContractSerializer(type, settings);
-            dcs.SetSerializationSurrogateProvider(SerializationSurrogateProvider);
+             if (Runtime.Fx.IsWasm && SerializationSurrogateProvider != null)
+              dcs.SetSerializationSurrogateProvider(SerializationSurrogateProvider);
```

Comment:
Import System.Runtime and refer to Fx.IsWasm instead of semi-qualifying the namespace. Also, SetSerializationSurrogateProvider has been supported since .NET Standard 1.6, is it really missing from Web Assembly? It was in an extension package in earlier versions.

Reply:
No, it is not missing but it triggers a NullReferenceException in mono when the parameter is null.
You could say that this is a bug in mono.